specs/topdown.md

```diff
@@ -0,0 +1,84 @@
+
+See [IPC Spec - IPLD Resolver](https://www.notion.so/IPC-Spec-IPLD-Resolver-7b4290a0d60c40cdba98cd6d3e66648b?pvs=21) for a more detailed discussion of the `VoteTally`.
+
+# Topdown Finality Proposal
+
+Once the `VoteTally` has detected a quorum, a parent finality proposal will be made. The entrypoint is this [link](https://github.com/consensus-shipyard/ipc/blob/7af25c4c860f5ab828e8177927a0f8b6b7a7cc74/fendermint/vm/topdown/src/finality/null.rs#L84). The high level idea for proposal creation is that it will first get the latest height in cache. However, do note that some fendermint nodes might be syncing faster than other fendermint nodes, if the proposed height is too large, other nodes will reject the proposal because they have yet to “see” it. As such, there is a `MaxProposalRange` parameter that controls how far the proposed height can be greater than the last committed block height. Once the height is set, the corresponding block hash is added to the proposal. The proposed topdown finality is added to the list of transactions sending to the cometbft.
+
+# Topdown Finality Proposal Validation
+
+Once a topdown finality proposal is received by a node, it’s [checked](https://github.com/consensus-shipyard/ipc/blob/7af25c4c860f5ab828e8177927a0f8b6b7a7cc74/fendermint/vm/interpreter/src/chain.rs#L209) to ensure its validity. The checking performs:
+
+- The proposed height either exists in topdown syncer
+- The block hash of the proposed height is correct.
+
+If the above two checks are valid, the proposal is accepted.
+
+# Topdown Finality Proposal Execution
+
+Once a proposal is accepted, it will be [executed](https://github.com/consensus-shipyard/ipc/blob/7af25c4c860f5ab828e8177927a0f8b6b7a7cc74/fendermint/vm/interpreter/src/chain.rs#L290) together with the batch of transactions. The execution of the proposal happens in three stages:
+
+- Commit new topdown finality
+- Fetch validator changes and store
+- Fetch topdown messages and execute
+
+The validator changes and topdown messages are first fetched from the parent syncer cache. If they are not present, then it will fallback to the RPC node. There is a simple exponential backoff [retry](https://github.com/consensus-shipyard/ipc/blob/7af25c4c860f5ab828e8177927a0f8b6b7a7cc74/fendermint/vm/topdown/src/finality/fetch.rs#L174). If the retry fails, which means the node cannot be reached to obtain the validator changes or topdown messages, an error is thrown. The reason for this is that the cluster has decided on the topdown finality, if one cannot obtain the corresponding changes and message, then it should stop.
+
+As topdown finality relies heavily on RPC node for querying, make sure the RPC node is reliable and returns enough historical data.
+
+See also [IPS Spec - Executions](https://www.notion.so/IPS-Spec-Executions-ebf13d833d6845ec9c11b59bd514fcda?pvs=21) for a description of how proposals and and executions are implemented.
```

Comment:
It shouldn't link to Notion anymore.

Reply:
the issue is currently there are no links to those specs yet, I will remove the hyper link first and add them back later.